src/leveled_log.erl

```diff
@@ -428,12 +432,15 @@ is_active_level([L|_], L, _) -> true;
 is_active_level([L|_], _, L) -> false;
 is_active_level([_|T], C, L) -> is_active_level(T, C, L).
 
--spec log_timer(atom(), list(), erlang:timestamp()) -> ok.
+-spec log_timer(atom(), list(), erlang:timestamp()) -> term().
 log_timer(LogReference, Subs, StartTime) ->
-    log_timer(LogReference, Subs, StartTime, ?LOG_LEVELS).
+    log_timer(LogReference, Subs, StartTime, ?LOG_LEVELS, ?LOGBASE, backend).
 
-log_timer(LogRef, Subs, StartTime, SupportedLevels) ->
-    {LogLevel, Log} = maps:get(LogRef, ?LOGBASE),
+-spec log_timer(
+    atom(), list(), erlang:timestamp(), list(log_level()), log_base(), atom())
+        -> term().
+log_timer(LogRef, Subs, StartTime, SupportedLevels, LogBase, Tag) ->
+    {LogLevel, Log} = maps:get(LogRef, LogBase, LogBase),
```

Comment:
This seems wrong. The default value for the get should either be omitted or be something computed from Supported levels or so. In any case not returning the `?LOGBASE` map and then matching that against a tuple.

Reply:
I think there should be no default (as with the log function).  If there was a default, it might have the wrong number of substitutions.  So we let it crash if there is a log to a missing log reference.

For leveled, and kv_index_tictactree they both have 100% code coverage in tests (and I maintain this as a requirement), and so there never should be a missing reference get through review.